packages/eslint-plugin/src/rules/explicit-function-return-type.ts

```diff
@@ -65,6 +70,17 @@ export default util.createRule<Options, MessageIds>({
       );
     }
 
+    /**
+     * Checks if the parent of a function expression has a type annotation.
+     * @param parent The parent of a function expression node
+     */
+    function hasTypeAnnotation(parent: TSESTree.Node): boolean {
```

Comment:
This function feels weird to me - it's name and signature are a bit disconnected from what it actually does.

The fact it is checking a "parent" is circumstantial, that fact is only true at call time - there is no hierarchy within the function, so the description and name don't feel right.

I think this fits better with what the function does:

```ts
function isVariableDeclaratorWithTypeAnnotation(node: TSESTree.Node): boolean
```

WDYT?

Reply:
That's also how I felt. I called the parameter `parent` to align with the conventions respected by the other functions from the rule. I'll also rename them for coherence then.